include/robot_interfaces/n_joint_robot_types.hpp

```diff
@@ -177,6 +178,53 @@ struct NJointRobotTypes
 
     typedef RobotFrontend<Action, Observation> Frontend;
     typedef std::shared_ptr<Frontend> FrontendPtr;
+
+
+    /**
+     * @brief Vector-YAML converter to implement automatic YAML conversion.
+     *
+     * To implement automatic YAML de- and encoding for a specific n-joint type,
+     * simply use the following code snippet:
+     *
+     *     struct MyTypes : public NJointRobotTypes<42> {};
+     *
+     *     namespace YAML
+     *     {
+     *     template <>
+     *     struct convert<MyTypes::Vector> : MyTypes::yaml_convert_vector {};
+     *     }
+     */
+    struct yaml_convert_vector
```

Comment:
i think this should probably not be here, since this is not specific to robot interfaces, and also the dependence on the template parameters can easily be avoided.
this should probably go to some generic package (e.g. mpi_cpp_tools)